src/pages/Home.js

```diff
@@ -1,9 +1,28 @@
 import React, { Component } from "react";
 import Join from "../components/Join";
 import Create from "../components/Create";
+import GetNewUser from "../components/GetNewUser";
 
 export default class Home extends Component {
+  constructor() {
+    super();
+    this.state = {
+      user: {}
+    };
+  }
+  componentDidMount() {
+    const userStr = localStorage.getItem("user");
+    if (userStr) {
+      const user = JSON.parse(userStr);
+      this.setState({ user });
+    }
+  }
+  updateUser(user) {
+    localStorage.setItem("user", JSON.stringify(user));
+    this.setState({ user });
```

Comment:
Getting `TypeError: Cannot read property 'setState' of undefined` on this line upon entering a username, probably just missing a `this` binding or change to arrow function.

Reply:
Everything should be working as expected now. Also added feature to edit username.